TorchSharp.PyBridge/PyBridgeModuleExtensions.cs

```diff
@@ -116,35 +114,107 @@ public static class PyBridgeModuleExtensions {
         /// <remarks>
         /// This method only supports loading the newer format used by `torch.save`, using a zip file. 
         /// The model will be fully loaded and all the validation checks will only run after the state
-        /// dictionary has been fully loaded. 
+        /// dictionary has been fully loaded.
         /// </remarks>
         public static Module load_py(this Module module, System.IO.Stream stream, bool strict = true, IList<string>? skip = null, Dictionary<string, bool>? loadedParameters = null, bool leaveOpen = false) {
-            // Create a dispose score so that we don't keep anyof the loaded tensors past this function
+            // Create a dispose score so that we don't keep any of the loaded tensors past this function
             using var d = torch.NewDisposeScope();
             using var d2 = torch.no_grad(); // To circumvent a bug introduced in 0.102.0
 
-            // Unpickle the state dictionary into memory
-            var stateHashtable = PyTorchUnpickler.UnpickleStateDict(stream, leaveOpen);
+            // Unpickle the state dictionary into memory.
+            // Keep stream open because tensors will not get deserialized yet.
+            var unpickled = PyTorchUnpickler.UnpickleStateDict(stream, leaveOpen: true, skipTensorRead: true);
 
             // Convert the hashtable to a dictionary of string->tensor
-            var stateDict = new Dictionary<string, torch.Tensor>();
-            foreach (string key in stateHashtable.Keys)
-                stateDict.Add(key, (torch.Tensor)stateHashtable[key]!);
+            Dictionary<string, PyTorchUnpickler.TensorConstructorArgs> unpickledConstructors = new();
```

Comment:
Style: The syntax `var variableName = new Type()` is preferred, in order to be consistent with the rest of the code.